aiohttp/client_exceptions.py

```diff
@@ -248,17 +249,34 @@ class InvalidURL(ClientError, ValueError):
 
     # Derive from ValueError for backward compatibility
 
-    def __init__(self, url: Any) -> None:
+    def __init__(self, url: StrOrURL, description: Union[str, None] = None) -> None:
         # The type of url is not yarl.URL because the exception can be raised
         # on URL(url) call
```

Comment:
Let's just store the params in attributes and avoid jumping through the hoops of accessing `self.args` + error handling in properties. Those would just return straightforward values.

```suggestion
        # on URL(url) call
        self._url = url
        self._description = description

```